mailbox/api/src/main/java/org/apache/james/mailbox/exception/ForbiddenDelegationException.java

```diff
@@ -19,5 +19,24 @@
 
 package org.apache.james.mailbox.exception;
 
+import org.apache.james.core.Username;
+
 public class ForbiddenDelegationException extends MailboxException {
+
+    Username givenUser;
+    Username delegateUser;
+
+    public ForbiddenDelegationException(Username givenUser, Username delegateUser) {
+        super("User " + givenUser.asString() + " is not delegated by " + delegateUser.asString());
+        this.givenUser = givenUser;
+        this.delegateUser = delegateUser;
+    }
+
+    public Username givenUser() {
+        return givenUser;
+    }
+
+    public Username delegateUser() {
+        return delegateUser;
+    }
```

Comment:
Where is this be used? I do not see one yet. Maybe you want to use it for tests in the next commits?

Reply:
IMO, it is a placeholder, We can easy get it when needed (maybe in feature)